internal/xds/rbac/converter.go

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright 2023 gRPC authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package rbac
+
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	v1xdsudpatypepb "github.com/cncf/xds/go/udpa/type/v1"
+	v3xdsxdstypepb "github.com/cncf/xds/go/xds/type/v3"
+	v3rbacpb "github.com/envoyproxy/go-control-plane/envoy/config/rbac/v3"
+	"google.golang.org/grpc/authz/audit"
+	"google.golang.org/protobuf/types/known/anypb"
+	"google.golang.org/protobuf/types/known/structpb"
+)
+
+func buildLogger(loggerConfig *v3rbacpb.RBAC_AuditLoggingOptions_AuditLoggerConfig) (audit.Logger, error) {
+	if loggerConfig.AuditLogger.TypedConfig == nil {
+		return nil, fmt.Errorf("AuditLogger TypedConfig cannot be nil")
```

Comment:
Does this get plumbed back to the user somehow?  I traced it back a couple levels and it's never augmented or changed.  I don't think this would be a useful error message for a user.  Similar with all the other errors in this function / PR - please make sure that the failure modes are understood and that the errors will make sense where they appear.

Reply:
It ends up bubbling up through the `NewEngine` func
Looks like that is called in `authz.NewStatic(authzPolicy string)` by the user or via xds config plumbing.

I think these errors make sense in the realm of the user getting them back from calling `authz.NewStatic(.), what do you think?

Reply:
The authz policy is a string and you're giving them errors about nil pointers.  That doesn't seem right.  `"missing require field: TypedConfig"` or something?

Reply:
That makes sense, I'll reword the errors to match that. Will comment back once I push that commit

Reply:
Reworded some, I think the others left still make sense?

Reply:
When the user calls `NewStatic`, are they ending up here?  I thought they wouldn't be using the xds protos in that case, and `TypedConfig` wouldn't even be a thing for them?

Reply:
The way the golang was already structured it all pushes into these protos in the end - the `rbac_engine.go` file is in `internal/xds` and was already written to use RBAC protos directly

Reply:
Does that mean this error won't ever go back out to users of `NewStatic` then, because we ensure the structure is right when we come through that path?

Reply:
In the `NewStatic` case it will have gone through `translatePolicy` in `rbac_translator.go`.
This converts from a string policy to the rbac proto.
Then that is passed along to `NewChainEngine`.

So checks that happen in `rbac_translator.go` will be guaranteed to be good by the time this path gets to `NewChainEngine`.